packages/gatsby/src/utils/worker/__tests__/jobs.ts

```diff
@@ -0,0 +1,237 @@
+import "jest-extended"
+import { createTestWorker, GatsbyTestWorkerPool } from "./test-helpers"
+import { store } from "../../../redux"
+import * as path from "path"
+import { waitUntilAllJobsComplete } from "../../jobs/manager"
+import type { MessagesFromChild, MessagesFromParent } from "../messaging"
+import { getReduxJobs, getJobsMeta } from "./test-helpers/child-for-tests"
+
+let worker: GatsbyTestWorkerPool | undefined
+
+describe(`worker (jobs)`, () => {
+  let mainProcessJobsMeta: ReturnType<typeof getJobsMeta>
+  let workersJobsMeta: Array<ReturnType<typeof getJobsMeta>>
+  let jobDescriptionsCreatedByWorkers: Array<Array<string>>
+  let mainStateAfter: ReturnType<typeof getReduxJobs>
+  let workerStateAfter: Array<ReturnType<typeof getReduxJobs>>
+  let sendMessageSpy: jest.SpyInstance<
+    void,
+    [msg: MessagesFromParent, workerId: number]
+  >
+  let receivedMessageSpy: jest.Mock<
+    void,
+    [msg: MessagesFromChild, workerId: number]
+  >
+
+  beforeAll(async () => {
+    store.dispatch({ type: `DELETE_CACHE`, cacheIsCorrupt: true })
+
+    worker = createTestWorker(3)
+
+    sendMessageSpy = jest.spyOn(worker, `sendMessage`)
+    receivedMessageSpy = jest.fn()
+    worker.onMessage(receivedMessageSpy)
+
+    const siteDirectory = path.join(__dirname, `fixtures`, `sample-site`)
+
+    await worker.all.loadConfigAndPlugins({
+      siteDirectory,
+      processFlags: false,
+    })
+
+    // plugin API creates a job
+    await Promise.all(worker.all.runAPI(`createSchemaCustomization`))
```

Comment:
Mixing between using `Promise.all` and not?

Reply:
made it consistent (and correct as await previously wasn't doing anything) in https://github.com/gatsbyjs/gatsby/pull/32162/commits/73a31872d379b1d8e9fd111e32a7b340b26a6ea5